packages/nx-go/src/go-package-graph/get-go-imports.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+type FileImportsInfo struct {
+	Path    string   `json:"path"`
+	Imports []string `json:"imports,"`
+}
+
+func main() {
+
+	if len(os.Args) == 1 {
+		println("no argument given, requires the directoy as an argument")
+		return
+	}
+
+	dirPath := os.Args[1]
+	info, err := os.Stat(dirPath)
+
+	if os.IsNotExist(err) || !info.IsDir() {
+		println("could not find directory:", dirPath)
+		return
+	}
+
+	goFiles, err := getGoFilesInDirectory(dirPath)
+
+	if err != nil {
+		println("bail out, error reading dir")
```

Comment:
**issue:** The error should be returned so it can help with diagnosing the issue.

```go
println("error reading dir: ". err)
```